File: mod16/main.rb
def count_unvisited_stones(stones, birds)
  visited = [false] * (stones + 1) # Создаем массив для отслеживания посещенных камней.
  birds.each do |bird| # Перебираем каждую птицу.
    (bird..stones).step(bird).each do |i| # Перебираем камни, которые может посетить птица.
      visited[i] = true # Помечаем камень как посещенный.
    end
  end
  (1..stones).count { |i| !visited[i] } # Подсчитываем количество непосещенных камней.
end

print "Введите количество камней: "
stones = gets.chomp.to_i # Получаем общее количество камней от пользователя.

print "Введите расстояния, которые могут преодолеть птицы (через пробел): "
birds = gets.chomp.split(" ").map(&:to_i) # Получаем расстояния, которые могут преодолевать птицы, от пользователя.

puts "Количество непосещенных камней: #{count_unvisited_stones(stones, birds)}" # Выводим результат.

File: mod16/Main.kt
fun countUnvisitedStones(stones: Int, birds: List<Int>): Int {
    val visited = BooleanArray(stones + 1) // Создаем массив для отслеживания посещенных камней.
    for (bird in birds) { // Перебираем каждую птицу.
        for (i in bird..stones step bird) { // Перебираем камни, которые может посетить птица.
            visited[i] = true // Помечаем камень как посещенный.
        }
    }
    return (1..stones).count { !visited[it] } // Подсчитываем количество непосещенных камней.
}

fun main() {
    print("Введите количество камней: ")
    val stones = readLine()!!.toInt() // Получаем общее количество камней от пользователя.

    print("Введите расстояния, которые могут преодолеть птицы (через пробел): ")
    val birds = readLine()!!.split(" ").map { it.toInt() } // Получаем расстояния, которые могут преодолевать птицы, от пользователя.

    println("Количество непосещенных камней: ${countUnvisitedStones(stones, birds)}") // Выводим результат.
}